spyder/plugins/projects/utils/watcher.py

```diff
@@ -87,10 +91,28 @@ def connect_signals(self, project):
         self.event_handler.sig_file_modified.connect(project.file_modified)
 
     def start(self, workspace_folder):
-        self.observer = Observer()
-        self.observer.schedule(
-            self.event_handler, workspace_folder, recursive=True)
-        self.observer.start()
+        # Needed to handle an error caused for inotify watch limit reached.
+        # See spyder-ide/spyder#10478
+        try:
+            self.observer = Observer()
+            self.observer.schedule(
+                self.event_handler, workspace_folder, recursive=True)
+            self.observer.start()
+        except OSError:
+            QMessageBox.warning(self.parent(), "Spyder",
```

Comment:
This seems unix specific?

Reply:
Yep

Reply:
Should we add a validation to only show the dialog for `linux` @goanpeca @ccordoba12 ?

Reply:
Should we add a validation to only show the dialog for `linux` @goanpeca @ccordoba12 ?

Reply:
Don't worry @dalthviz, there's no need of that because `inotify` only works on Linux.